### PR TITLE
Review suggestions for #2181

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
- We cannot use a ratio to decide if the LMBD db should be resized with each write, instead, we need a to allow the db size to grow up to the resize threshold (environment size - current size < threshold size) and then resize it once with the init size.
- Simplified resize logic and functions for startup and write operations.

## Motivation and Context
- As implemented the db resized continuously.
- The code could be simplified for startup and write operations, as it is not required to test if resize is required in one function and then resize in another function.

## How Has This Been Tested?
Tested with a base node sync from scratch in Windows 10.

## Types of changes

* [X] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [ ] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: infrastructure/storage/src/lmdb_store/store.rs [hunk 1]
```diff
@@ -375,48 +359,27 @@ impl LMDBStore {
...
-        let previous_size = env.info()?.mapsize;
-
-        unsafe {
-            env.set_mapsize(new_size_bytes)?;
+    /// Resize the LMDB environment if the resize threshold is breached.
+    pub fn resize_if_required(env: &Environment, init_size_bytes: usize) -> Result<(), LMDBError> {
+        let env_info = env.info()?;
+        let size_used_bytes = env.stat()?.psize as usize * env_info.last_pgno;
+        let needs_resize = env_info.mapsize - size_used_bytes <= (init_size_bytes as f64 * RESIZE_THRESHOLD) as usize;
+        let path = env.path().unwrap().to_str().unwrap();
```
Comment (hansieodendaal) [inline infrastructure/storage/src/lmdb_store/store.rs:367]:
```suggestion
```

Comment (hansieodendaal) [reply infrastructure/storage/src/lmdb_store/store.rs:367]: Realized this is an unnecessary call each time, could be simplified

File: infrastructure/storage/src/lmdb_store/store.rs [hunk 2]
```diff
@@ -375,48 +359,27 @@ impl LMDBStore {
...
+        if needs_resize {
+            unsafe {
+                env.set_mapsize(size_used_bytes + init_size_bytes)?;
+            }
+            debug!(
+                target: LOG_TARGET,
+                "({}) LMDB size used {:?} MB, environment space left {:?} MB, increased by {:?} MB.",
+                path,
```
Comment (hansieodendaal) [inline infrastructure/storage/src/lmdb_store/store.rs:375]:
```suggestion
                env.path().unwrap().to_str().unwrap(),
```

Comment (hansieodendaal) [reply infrastructure/storage/src/lmdb_store/store.rs:375]: As per above

**Review by delta1 — APPROVED** [review]

**Comment by hansieodendaal** [comment]
Sync from scratch log file entries with above code change suggestions:
```
2020-09-06 18:19:53.363100300 [lmdb] INFO  (rincewind\db) LMDB environment created with a capacity of 1000 MB, 999 MB remaining, resize threshold is 50.00 %.
2020-09-06 18:19:53.366552400 [lmdb] TRACE (rincewind\db) LMDB database 'kernels' is ready
2020-09-06 18:19:53.369093600 [lmdb] TRACE (rincewind\db) LMDB database 'range_proof_mmr_cp_backend' is ready
2020-09-06 18:19:53.370209500 [lmdb] TRACE (rincewind\db) LMDB database 'metadata' is ready
2020-09-06 18:19:53.372252500 [lmdb] TRACE (rincewind\db) LMDB database 'kernel_mmr_cp_backend' is ready
2020-09-06 18:19:53.374247300 [lmdb] TRACE (rincewind\db) LMDB database 'block_hashes' is ready
2020-09-06 18:19:53.376222500 [lmdb] TRACE (rincewind\db) LMDB database 'stxos' is ready
2020-09-06 18:19:53.377265200 [lmdb] TRACE (rincewind\db) LMDB database 'orphans' is ready
2020-09-06 18:19:53.379220800 [lmdb] TRACE (rincewind\db) LMDB database 'headers' is ready
2020-09-06 18:19:53.381268700 [lmdb] TRACE (rincewind\db) LMDB database 'txos_hash_to_index' is ready
2020-09-06 18:19:53.382261200 [lmdb] TRACE (rincewind\db) LMDB database 'utxos' is ready
2020-09-06 18:19:53.384226800 [lmdb] TRACE (rincewind\db) LMDB database 'utxo_mmr_cp_backend' is ready
2020-09-06 18:19:53.575266500 [lmdb] INFO  (rincewind\peer_db) LMDB environment created with a capacity of 50 MB, 49 MB remaining, resize threshold is 50.00 
2020-09-06 18:19:53.577281500 [lmdb] TRACE (rincewind\peer_db) LMDB database 'peers' is ready
2020-09-06 18:19:53.857248400 [lmdb] INFO  (rincewind\wallet_peer_db) LMDB environment created with a capacity of 50 MB, 49 MB remaining, resize threshold is
2020-09-06 18:19:53.859248900 [lmdb] TRACE (rincewind\wallet_peer_db) LMDB database 'peers' is ready
2020-09-06 23:57:37.728888700 [lmdb] DEBUG (rincewind\db) LMDB size used 502 MB, environment space left 497 MB, increased by 1000 MB.
2020-09-07 04:34:13.911425000 [lmdb] DEBUG (rincewind\db) LMDB size used 1002 MB, environment space left 499 MB, increased by 1000 MB.
```

**Pull request merged**